contracts/standard/arbitration/MultipleArbitrableTransaction.sol

```diff
@@ -0,0 +1,320 @@
+     */
+    function timeOutByBuyer(uint _transactionId) {
+        Transaction storage transaction = transactions[_transactionId];
+        require(msg.sender == transaction.buyer);
+
+
+        require(transaction.status == Status.WaitingSeller);
+        require(now>=transaction.lastInteraction+transaction.timeout);
+        
+        executeRuling(transaction.disputeId,BUYER_WINS);
+    }
+    
+    /** @dev Submit a reference to evidence. EVENT.
+     *  @param _transactionId The index of the transaction.
+     *  @param _evidence A link to an evidence using its URI.
+     */
+    function submitEvidence(uint _transactionId, string _evidence) {
+        Transaction storage transaction = transactions[_transactionId];
+        require(msg.sender == transaction.buyer || msg.sender == transaction.seller);
+        
+        require(transaction.status>=Status.DisputeCreated);
+        Evidence(_transactionId, transaction.arbitrator,transaction.disputeId,msg.sender,_evidence);
+    }
+    
+    /** @dev Appeal an appealable ruling.
+     *  Transfer the funds to the arbitrator.
+     *  Note that no checks are required as the checks are done by the arbitrator.
+     *  @param _transactionId The index of the transaction.
+     *  @param _extraData Extra data for the arbitrator appeal procedure.
+     */
+    function appeal(uint _transactionId, bytes _extraData) payable public {
+        Transaction storage transaction = transactions[_transactionId];
+        require(msg.sender == transaction.buyer || msg.sender == transaction.seller);
+        
+        transaction.arbitrator.appeal.value(msg.value)(transaction.disputeId,_extraData);
+    }
+    
+
+
+    /** @dev
+     *  @param _arbitrator The arbitrator of the contract.
+     *  @param _hashContract Keccak hash of the plain English contract.
+     *  @param _timeout Time after which a party automatically loose a dispute.
+     *  @param _seller The recipient of the transaction.
+     *  @param _arbitratorExtraData Extra data for the arbitrator.
+     */
+    function createTransaction(Arbitrator _arbitrator, bytes32 _hashContract, uint _timeout, address _seller, bytes _arbitratorExtraData) payable public returns (uint transactionIndex) {
+        transactions.push(Transaction({ 
+            seller: _seller,
+            buyer: msg.sender,
+            amount: msg.value,
+            timeout: _timeout,
+            disputed: false,
+            arbitrator: _arbitrator,
+            arbitratorExtraData: _arbitratorExtraData,
+            disputeId: 0,
+            sellerFee: 0,
+            buyerFee: 0,
+            lastInteraction: now,
+            status: Status.NoDispute
+        }));
+        ContractHash(transactions.length - 1, _hashContract);
+        return transactions.length - 1;
+    }
+
+    /** @dev
+     * returns the last transaction id whose buyer is the sender
+     */
+    function lastTransactionId () constant public returns (uint index) {
```

Comment:
This has a O(n) complexity and thus should only be used by the UI (we need to warn about that).
I also don't think this function is required.

Reply:
@clesaege I only use this in tests. Can I keep it? :)

Reply:
Do you have a way to do without it? If so that would be better in order to lower code complexity (code ,complexity in tests is better than code complexity in smart contracts).

Reply:
Yes, I'll figure it out